staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -275,7 +282,287 @@ func TestDecode(t *testing.T) {
+		},
+		{
+			name: "map with different key types",
+			in:   hex("a201616161626132"), // { 1: "a", "b": "2"}
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "positive integer" || e.GoType != "string" {
+					t.Errorf("Expected types: positive integer,string. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name: "map with non string key types",
+			in:   hex("a1fb40091eb851eb851f63706965"), // { 3.14: "pie"}
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "primitives" || e.GoType != "string" {
+					t.Errorf("Expected types: primitives,string. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name:          "map with byte string key type",
+			in:            hex("a143abcdef187b"), // {h'ABCDEF': 123}
+			want:          map[string]interface{}{"\xab\xcd\xef": int64(123)},
+			assertOnError: assertNilError,
+		},
+		{
+			name:  "max positive integer value supported by cbor: 2^64 - 1",
+			modes: []cbor.DecMode{modes.Decode},
+			in:    hex("1bffffffffffffffff"), // 18446744073709551615
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "positive integer" || e.GoType != "int64" {
+					t.Errorf("Expected types: positive integer,int64. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name: "min negative integer value supported by cbor: -2^64",
+			in:   hex("3bffffffffffffffff"), // -18446744073709551616
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "negative integer" || e.GoType != "int64" {
+					t.Errorf("Expected types: negative integer,int64. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name:          "tag 0 timestamp string",
+			in:            hex("c074323030362d30312d30325431353a30343a30355a"), // 0("2006-01-02T15:04:05Z")
+			want:          "2006-01-02T15:04:05Z",
+			failureReason: "decoding cbor data tagged with 0 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:            "tag 0 with byte string",
+			in:              hex("c043414243"), // 0('ABC')
+			unexportedError: "cbor: tag number 0 must be followed by text string, got byte string",
+		},
+		{
+			name:            "tag 0 with invalid text string",
+			in:              hex("c06474657874"), // 0("text")
+			unexportedError: "cbor: cannot set text for time.Time: parsing time",
+		},
+		{
+			name:          "tag 1 timestamp unsigned integer",
+			in:            hex("c11a43b940e5"), // 1(1136214245)
+			want:          "2006-01-02T15:04:05Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with float16 value",
+			in:            hex("c1f93c00"), // 1(1.0)
```

Comment:
Might as well use the special encoding indicators in these three comments to make it clear which float width is being used.

```suggestion
			in:            hex("c1f93c00"), // 1(1.0_1)
```

Reply:
Fixed.